### PR TITLE
SQL: `RANDOM(<expr>)` always evaluates to `NULL` if `<expr>` is `NULL`

### LINKED ISSUE #84627: SQL: `RANDOM(null)` is not always `null`
### Elasticsearch Version

8.1.0-SNAPSHOT (but also affects 7.x releases)

### Installed Plugins

_No response_

### Java Version

_bundled_

### OS Version

OSX

### Problem Description

While the docs state that `RANDOM(null) = null` that's not always the case.

### Steps to Reproduce

```
GET _sql?format=txt
{
  "query": """
  SELECT random(n), n FROM longs
  """
}
```

yields 

```
    random(n)     |       n       
------------------+---------------
0.7308781907032909|1              
0.7311469360199058|2              
0.3860252809814635|null           
```

Expected would be `null` for the last row.

### Logs (if relevant)

_No response_

### PR DESCRIPTION
**Pull request by Luegg**
Resolves #84627

### CODE REVIEW

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/math/MathProcessor.java [hunk 1]
```diff
@@ -77,7 +76,7 @@
         LOG10(Math::log10),
         PI(() -> Math.PI),
         RADIANS(Math::toRadians),
-        RANDOM((Object l) -> l != null ? new Random(((Number) l).longValue()).nextDouble() : Randomness.get().nextDouble(), true),
```
Comment (astefan) [inline x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/math/MathProcessor.java:80]: This was the only place where the `true` value for `nullAware` was used. Maybe it's worth adjusting the constructors as well.

**Review by astefan — APPROVED** [review]
LGTM. Left one comment.

**Review by luigidellaquila — APPROVED** [review]
LGTM

File: x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/math/MathProcessor.java (hunk 1 above)
Comment (Luegg) [reply x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/math/MathProcessor.java:80]: good point, I removed the ctor 👍

**Review by costin — APPROVED** [review]
LGTM code wise however I want to clarify the behavior since this is a breaking change.
Despite was the docs said, previously null was considered as no seed e.g. random() ; now it nullifies the output.
I think a follow-up is to allow random to be called without any argument (no seed).

Btw, are there any benefits of using Random instead of Randomness - if we give the same seed (such as the hashcode of a long lived/static object such as hashcode of a string constant or PlanExecutor) the queries would returned reproduceable numbers for the same seed on the same node.
Maybe there are others values/statics that we can use across node per cluster lifecycle so that the same query (random(1)) returns the same result regardless of the node use as long as the cluster is not being restarted.

**Review by bpintea — APPROVED** [review]

**Pull request merged**